lib/internal/errors.js

```diff
@@ -111,7 +111,23 @@ module.exports = exports = {
 //
 // Note: Please try to keep these in alphabetical order
 E('ERR_ARG_NOT_ITERABLE', '%s must be iterable');
+E('ERR_ARG_OUT_OF_BOUNDS',
+  (name) => `The "${name}" argument is out of bounds`);
 E('ERR_ASSERTION', (msg) => msg);
+E('ERR_BUFFER_ENCODING_PROVIDED',
+  'If encoding is specified then the first argument must be a string');
+E('ERR_BUFFER_INVALID_CONCAT_LIST',
+  'The "list" argument must be an Array of Buffer or Uint8Array instances');
+E('ERR_BUFFER_TRANSCODE_FAILED',
+  (code) => `Unable to transcode Buffer [${code}]`);
+E('ERR_BUFFER_MISALIGN',
+  (size) => `Buffer size must be a multiple of ${size}-bits`);
+E('ERR_BUFFER_OVERFLOW',
+  (max) => `The "size" argument must not be greater than ${max}`);
+E('ERR_BUFFER_UNDERFLOW', 'The "size" argument must not be negative');
```

Comment:
Maybe call the error type something more generic? There could be any variable that may not be negative or that may not be greater than x.